src/simulator/world.py

```diff
@@ -170,7 +172,18 @@ def visualize(self) -> dict:
         """
         objects_info = dict()
         for oid in self.objects:
-            if self.objects[oid].shape is not None:
                 # TODO: to be replaced with proper handling of compound objects (if cond)
-                objects_info[oid] = self.objects[oid].visualize()
+            objects_info[oid] = self.objects[oid].visualize()
         return objects_info
+    
+    def dumb_all_owners_inf(self):
+        # dump visualization info for Owners between objects to the output json file
+        owners_info = {"Owners":{}}
+        for Oid in self.objects:
+            if self.objects[Oid].owner_object is None:
+                if len(self.objects[Oid].dependent_objects) == 0 :
+                    owners_info["Owners"][Oid] = None
+                else:
+                    owners_info["Owners"][Oid] = list(map(str,list(self.objects[Oid].dependent_objects.keys())))
```

Comment:
How does this piece of code work? For nested dependencies (i.e. A owns B, and B owns C), what are we storing as the owner for B and C?

Reply:
Ditto as above, please see my comment on the proposal document.